src/scales.js

```diff
@@ -105,3 +116,22 @@ function inferScaleType(key, channels, {type, domain, range}) {
 function asOrdinalType(key) {
   return registry.get(key) === position ? "point" : "ordinal";
 }
+
+// prepare scales for exposure through the plot's scales() function
+export function exposeScales(scaleDescriptors) {
```

Comment:
It’d probably be nice if *plot*.scales was implemented as a lazy getter so that we don’t need to materialize these copies in the vast majority of cases where they won’t be needed. That can be done by using Object.defineProperty to define a getter on the return plot element, but we’ll also want to make that getter replace itself with the materialized value if it’s called more than once so that we don’t create a new object each time.

Reply:
I implemented each of scales[key] as an independent getter with a cache. First time I used Object.defineProperty :)